src/lib/optimizer/strategy/in_expression_rewrite_rule.cpp

```diff
@@ -111,6 +113,8 @@ void InExpressionRewriteRule::_apply_to_plan_without_subqueries(
     return;
   }
 
+  const auto cardinality_estimator = cost_estimator->cardinality_estimator->new_instance();
```

Comment:
(on mobile)
Is that really “for free”?

Reply:
Sorry, my bad, I thought that we initialize the cardinality estimator only if we have an `IN` statement but we now do that beforehand.

However, the queries suffering the most are the long-running queries. I would expect optimizer costs to have a larger impact on the short running queries. Still, I could move the instantiation of the cardinality estimator down to the part where we already ensured that we have an `IN` statement.